crates/noirc_frontend/src/ast/type_alias.rs

```diff
@@ -0,0 +1,33 @@
+use crate::{Ident, UnresolvedGenerics, UnresolvedType};
+use iter_extended::vecmap;
+use noirc_errors::Span;
+use std::fmt::Display;
+
+/// Ast node for type aliases
+#[derive(Clone, Debug)]
+pub struct NoirTyAlias {
+    pub name: Ident,
+    pub generics: UnresolvedGenerics,
+    pub ty: UnresolvedType,
+    pub span: Span,
+}
+
+impl NoirTyAlias {
```

Comment:
Can we just call this `NoirTypeAlias`? I don't feel the need to shorten the name as it is not that long and clearer